integration/fetcher-state-test.ts

```diff
@@ -0,0 +1,271 @@
+                  </button>
+                </fetcher.Form>
+                <button id="${TYPES.normalLoad}" onClick={() => fetcher.load('/page')}>
+                  Submit ${TYPES.normalLoad}
+                </button>
+              </>
+            );
+          }
+        `,
+        "app/routes/redirect.jsx": js`
+          import { useFetcher } from "@remix-run/react";
+          export function loader() {
+            return { from: 'redirect loader' }
+          }
+          export default function() {
+            return <h1>Redirect</h1>;
+          }
+        `,
+      },
+    });
+
+    appFixture = await createAppFixture(fixture, ServerMode.Development);
+  });
+
+  test.afterAll(async () => {
+    await appFixture.close();
+  });
+
+  test("represents a initial fetcher", async ({ page }) => {
+    let app = new PlaywrightFixture(appFixture, page);
+    await app.goto("/page", true);
+    let text = (await app.getElement("#initial-state")).text();
+    expect(JSON.parse(text)).toEqual({
+      state: "idle",
+      type: "init",
+    });
+  });
+
+  test("represents an actionSubmission fetcher", async ({ page }) => {
+    let app = new PlaywrightFixture(appFixture, page);
+    await app.goto("/page", true);
+    await app.clickElement(`#${TYPES.actionSubmission}`);
+    await page.waitForSelector("#loading", { state: "hidden" });
+    let text = (await app.getElement("#states")).text();
+    expect(JSON.parse(text)).toEqual([
+      {
+        state: "submitting",
+        type: "actionSubmission",
+        submission: {
+          formData: { key: "value" },
+          action: "/page",
+          method: "POST",
+          encType: "application/x-www-form-urlencoded",
+        },
+      },
+      {
+        state: "loading",
+        type: "actionReload",
+        submission: {
+          formData: { key: "value" },
+          action: "/page",
+          method: "POST",
+          encType: "application/x-www-form-urlencoded",
+        },
+        data: {
+          from: "action",
+        },
+      },
+      {
+        state: "idle",
+        type: "done",
+        data: {
+          from: "action",
+        },
+      },
+    ]);
+  });
+
+  test("represents a loaderSubmission fetcher", async ({ page }) => {
+    let app = new PlaywrightFixture(appFixture, page);
+    await app.goto("/page", true);
+    await app.clickElement(`#${TYPES.loaderSubmission}`);
+    await page.waitForSelector("#loading", { state: "hidden" });
+    let text = (await app.getElement("#states")).text();
+    expect(JSON.parse(text)).toEqual([
+      {
+        state: "submitting",
+        type: "loaderSubmission",
+        submission: {
+          formData: { key: "value" },
+          // TODO: This feels potentially wrong - the formAction is just /page,
+          // the request URL is /page?key=value
+          action: "/page?key=value",
+          method: "GET",
+          encType: "application/x-www-form-urlencoded",
+        },
```

Comment:
@jacob-ebey What do you think here?  IIRC we specifically tested against this in `@remix-run/router` so the `submission` stayed true to the state of the DOM and only the `request` represented the serialized `URLSearchParams`

Reply:
Same behavior happens for `loaderSubmission` transitions - updated the test in `transition-state-test` below with the same logic